### PR TITLE
IMPROVE - OdometerView Preview with Mock Data

### LINKED ISSUE #347: IMPROVE - `OdometerView` Preview with Mock Data
## Describe the feature you want to improve
`OdometerView` shows the empty view as the preview, but it would be awesome to somehow pass the proper data, and/or mock it, so that we actually see example rows of data, rather than the empty screen when looking at the previews

I'm not sure what this will take, and hopefully it doesn't require too much of moving things around, but this will also help with unit testing, I'm guessing

### Empty view, in previews
Screenshot 2024-10-21 at 1 15 36 AM
### screenshot of the current app when you run it, and add odometer readings
I'd like to see something like this in the previews
Simulator Screenshot - iPhone 16 Pro - 2024-10-21 at 01 15 39

## Describe alternatives you've considered

### PR DESCRIPTION
**Pull request by mike011**
# What it Does
* Closes #347 

- Exposed the viewModel to the file via a `fileprivate init`.
- In the Preview section injected the example data into the `viewModel`. 

# How I Tested
The preview panel is showing the example as laid out in the issue.

# Notes

# Screenshot
image

### CODE REVIEW

File: Basic-Car-Maintenance/Shared/Odometer/Views/OdometerView.swift
Comment (mikaelacaron) [inline Basic-Car-Maintenance/Shared/Odometer/Views/OdometerView.swift:142]: I moved these inside the `#Preview` because they'd be local to that scope by default

File: Basic-Car-Maintenance/Shared/Odometer/Views/OdometerView.swift
Comment (mikaelacaron) [inline Basic-Car-Maintenance/Shared/Odometer/Views/OdometerView.swift:116]: I moved some formatting, as opposed to appending everything individually, you can append them all together as an array

**Review by mikaelacaron — APPROVED** [review]
Great work!!

**Comment by mikaelacaron** [comment]
@mike011 can you please go back and check a checkbox that says the maintainer can edit the PR, because otherwise I can't make the commit that I just did. (Click the 3 dots on the PR description, I think, and there somewhere should be a checkbox)

(I could ask you do to the changes that I noted, but for something small like this it's easier for me to edit myself, and immediately merge your code, rather than waiting for you to make the changes, and me having to re-review it.

After this, I can actually merge your code

Screenshot 2024-10-27 at 4 50 15 PM

**Comment by mike011** [comment]
I couldn’t find where to give you permissions so I opened a new PR with it turned on.

**Pull request closed without merging**
